File: sdk/java-guest/examples/service.java
package dev.latent.app;
import dev.latent.generated.Bindings;
import dev.latent.generated.Bindings.*;
import dev.latent.guest.*;
import java.nio.charset.StandardCharsets;
import java.util.Arrays;
import java.util.List;

public final class Capsule implements Bindings.Exports {
    public Unsigned64 run(Long which, String text, Unsigned64 handle) {
        var target = new LatentServiceInvokeTarget(Option.none(), "callee", "tests:local/api@1.0.0",
            which == 0 ? "answer" : which == 1 ? "fail" : "spin", Option.some("callee"));
        var result = LatentServiceInvoke.call(target, "[]".getBytes(StandardCharsets.UTF_8),
            "application/vnd.latent.wit-values.v1+json", new LatentServiceInvokeCallOptions(Option.none(), (short) 0, Option.none(), List.of()));
        if (result.tag() == 0) {
            if (!Arrays.equals(result.successValue().payload(), "[42]".getBytes(StandardCharsets.UTF_8))) throw new IllegalStateException("callee payload mismatch");
            return new Unsigned64(42);
        }
        if (result.tag() == 1) {
            if (result.declaredErrorValue().payload().length == 0) throw new IllegalStateException("missing declared error");
            return new Unsigned64(10);
        }
        return new Unsigned64(switch (result.platformFailureValue().code()) {
            case PermissionDenied -> 11; case Cancelled -> 12; case DeadlineExceeded -> 13; case ResourceExhausted -> 14;
            default -> throw new IllegalStateException("unexpected service platform failure");
        });
    }
}

File: sdk/typescript-guest/examples/service.ts
import type * as Contract from '../generated/interfaces/tests-caller-api.js';
import { call } from '../vendor/lsf/sdk/typescript-guest/capabilities/service.js';
export const api: typeof Contract = {
  run(which) {
    const result = call({ service: 'callee', contract: 'tests:local/api@1.0.0',
      function: which === 0 ? 'answer' : which === 1 ? 'fail' : 'spin', route: 'callee' },
    new Uint8Array([91, 93]), 'application/vnd.latent.wit-values.v1+json', { priority: 0, metadata: [] });
    if (result.tag === 'declared-error') {
      if (result.val.payload.length === 0) throw new Error('missing-declared-payload');
      return 10n;
    }
    if (result.tag === 'platform-failure') {
      switch (result.val.code) {
        case 'permission-denied': return 11n;
        case 'cancelled': return 12n;
        case 'deadline-exceeded': return 13n;
        case 'resource-exhausted': return 14n;
        default: throw new Error('unexpected-child-failure');
      }
    }
    if (result.val.payload.length !== 4 || result.val.payload.some((byte, i) => byte !== [91, 52, 50, 93][i]))
      throw new Error('child-payload-mismatch');
    return 42n;
  },
};
